### PR TITLE
Pv2 1163 data lock status changed service

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.DataLocks.Domain/Services/DataLockStatusService.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                return DataLockStatusChange.NoChange;
+
+            if (oldFailures == null)
+                return DataLockStatusChange.ChangedToFailed;
+
+            if (newFailures == null)
+                return DataLockStatusChange.ChangedToPassed;
+
+            if (oldFailures.Count != newFailures.Count)
+                return DataLockStatusChange.FailureCodeChanged;
```
Comment (carlbm) [inline src/SFA.DAS.Payments.DataLocks.Domain/Services/DataLockStatusService.cs]: Can you change the name of `FailureCodeChanged` - it's a bit confusing

File: src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockEventProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                LearningAimFrameworkCode = dataLockEvent.LearningAim.FrameworkCode,
+                LearningAimPathwayCode = dataLockEvent.LearningAim.PathwayCode,
+                LearningAimProgrammeType = dataLockEvent.LearningAim.ProgrammeType,
+                LearningAimReference = dataLockEvent.LearningAim.Reference,
+                LearningAimStandardCode = dataLockEvent.LearningAim.StandardCode,
+                Errors = failures
+            };
+        }
+
+        private static void AddTypeAndPeriodToEvent(DataLockStatusChanged changedToPassed, TransactionType transactionType, byte period)
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockEventProcessor.cs]: corrected

File: src/SFA.DAS.Payments.DataLocks.Domain/Services/DataLockStatusService.cs (hunk 1 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.Domain/Services/DataLockStatusService.cs]: renamed

File: src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockEventProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                    }
+                }
+                
+            }
+
+            foreach (var incentiveEarning in dataLockEvent.IncentiveEarnings)
+            {
+                foreach (var period in incentiveEarning.Periods)
+                {
+                    if (result.TryGetValue(((TransactionType) incentiveEarning.Type, period.Period), out var failures))
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockEventProcessor.cs]: can't we just say if (!result.TryGetValue) as failures are not been used in the outgoing output from this method

Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockEventProcessor.cs]: `failures` are part of the `result` collection that is being returned from the method

**Comment by kuzinator** [comment]
This has been approved and merged into DataLockEvents feature branch.

**Pull request closed without merging**